### DISCUSSION TITLE
[4.2] change preprocessForm to support plugin group

### DISCUSSION THREAD

**Discussion by Flowman** [joomla/joomla-cms#47112] [2. Cold Features]
### Steps to reproduce the issue

Right now we can inject a custom form into preprocessForm with onContentPrepareForm event from a plugin group.

Issue is when trying to save the form it does not inject the plugin group name into preprocessForm and the controller validation remove the custom data entered into the form data.

This line should be extended with the plugin group value of the preprocessForm function. As right now it will always load the 'content' group and there is no way to override it.

https://github.com/joomla/joomla-cms/blob/1d3ca40ef6c55b2de63c40410a96af695a90a7cf/libraries/src/MVC/Model/FormBehaviorTrait.php#L107

### Expected result

Be able to load plugin group for preprocessForm  on save.

### Actual result

Not possible.

### System information (as much as possible)

Joomla 4.2

### Additional comments

Maybe add an $options['group'] as the would be hard to extend the loadForm function with a new parameter. 

Example:
$this->preprocessForm($form, $data, $options['group']);

**Comment by Fedik** [comment]
Sounds valid.

Note: Currently if you use own model, you can override preprocessForm and load your plugin group  
https://github.com/joomla/joomla-cms/blob/1d3ca40ef6c55b2de63c40410a96af695a90a7cf/libraries/src/MVC/Model/FormBehaviorTrait.php#L163-L170

```php
 PluginHelper::importPlugin('my-plugin-group');
```

**Comment by Flowman** [comment]
True didn't think of that. But would be nice if core fixed it out of the box.
